packages/cli/src/utils/plugins.ts

```diff
@@ -0,0 +1,317 @@
+
+      const pageFile = paths.pop()
+      const pagePaths = paths
+
+      const pagePath = path.join(tmpPagesDir, ...pagePaths, pageFile + '.tsx')
+
+      const fileContent = getPluginPageFileContent(
+        sanitizePluginName(pluginName),
+        pageName,
+        pagesConfig[pageName].appLayout
+      )
+
+      mkdirSync(path.dirname(pagePath), { recursive: true })
+      writeFileSync(pagePath, fileContent)
+    })
+  })
+}
+
+export async function addPluginsSections(basePath: string, plugins: Plugin[]) {
+  const { tmpPluginsDir, getPackagePath } = withBasePath(basePath)
+
+  logger.log('Adding plugin sections')
+
+  const indexPluginsOverrides = plugins
+    .filter((plugin) =>
+      existsSync(
+        getPackagePath(getPluginName(plugin), 'src', 'components', 'index.ts')
+      )
+    )
+    .map((plugin) => {
+      const pluginReference =
+        sanitizePluginName(getPluginName(plugin), true) + 'Components'
+
+      return {
+        import: `import { default as ${pluginReference} } from 'src/plugins/${sanitizePluginName(getPluginName(plugin))}/components'`,
+        pluginReference,
+      }
+    })
+
+  const pluginsImportFileContent = `
+  ${indexPluginsOverrides.map((plugin) => plugin.import).join('\n')}
+
+  export default {
+    ${indexPluginsOverrides.map((plugin) => `...${plugin.pluginReference}`).join(',\n')}
+  }
+  `
+
+  const sectionPath = path.join(tmpPluginsDir, 'index.ts')
+  writeFileSync(sectionPath, pluginsImportFileContent)
+  logger.log('Writing plugins overrides')
+  logger.log(sectionPath)
+  logger.log(pluginsImportFileContent)
+}
+
+export async function addPluginsOverrides(basePath: string, plugins: Plugin[]) {
+  const { tmpPluginsDir, getPackagePath } = withBasePath(basePath)
+
+  logger.log('Adding plugin overrides')
+
+  plugins
+    .map((plugin) => ({
+      pluginName: getPluginName(plugin),
+      pluginOverridesPath: getPackagePath(
+        getPluginName(plugin),
+        'src',
+        'components',
+        'overrides'
+      ),
+    }))
+    .filter(({ pluginOverridesPath }) => existsSync(pluginOverridesPath))
+    .reverse()
+    .forEach(({ pluginName, pluginOverridesPath }) => {
+      const overrideFilesAlreadyCopied: string[] = []
+
+      const sanitizedPluginName = sanitizePluginName(pluginName)
+
+      const overrideFiles = readdirSync(pluginOverridesPath)
+
+      overrideFiles
+        .filter((file) => !overrideFilesAlreadyCopied.includes(file))
+        .forEach((overrideFileName) => {
+          const overrideFileContent = `export { override } from 'src/plugins/${sanitizedPluginName}/components/overrides/${overrideFileName.split('.')[0]}'`
+
+          writeFileSync(
+            path.join(tmpPluginsDir, 'overrides', overrideFileName),
+            overrideFileContent
+          )
+          overrideFilesAlreadyCopied.push(overrideFileName)
+        })
+    })
+}
+
+export const generateThemeIndexPluginsContent = async (
```

Comment:
Looks like this function is not being called elsewhere. Should we remove or do you plan to use it in the future?

Reply:
It's truth! Removed, thank you!